core/storage/exploration/gae_models.py

```diff
@@ -865,6 +870,123 @@ def export_data(cls, user_id: str) -> Dict[str, List[str]]:
         }
 
 
+class TransientCheckpointUrlModel(base_models.BaseModel):
```

Comment:
Regarding decoding the unique_id -- 
1. It would actually be quite difficult to encode so many fields in a single string.
2. Since the information related to the user progress would change at every checkpoint, the unique_id (and also the asscoiated url) would have to be altered everytime (which is supposed to be unique for each logged-out learner)

Reply:
Ah, sorry I forgot to reply.

1. I don't think it is that hard but the string would be much longer than now.
2. Yep, agreed but generating a new string is quite fine compared to always changing the datastore.

The ID of the model is now 6 characters is this for some reason? Like will the user need to remember the code or will it just be saved in the browser? If it will just be saved then I think it is fine to have much longer string.

Reply:
Sorry for the late reply -- 
1. Currently, the learner has to remember the code and use the same code throughout his/her journey as a logged-out learner in that exploration (upto 72 hours (After the cron job)).
2. Since this code will be the part of the url that the learner would put in to get back to the last visited checkpoint, we actually wanted it to be a bit handy.
3. Afterwards, we are thinking to store even the learner's answers, so, we would need a storage model anyway.